File: Lab11/app/src/main/java/com/example/lab11/MainActivity.java
package com.example.lab11;

import androidx.appcompat.app.AppCompatActivity;

import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private EditText ed_book,ed_price;
    private Button btn_query,btn_insert,btn_update,btn_delete;
    private ListView listView;
    private ArrayAdapter<String> adapter;
    private ArrayList<String> items =new ArrayList<>();
    private SQLiteDatabase dbrw;

    @Override
    protected void onDestroy(){
        super.onDestroy();
        dbrw.close();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ed_book=findViewById(R.id.ed_book);
        ed_price=findViewById(R.id.ed_price);
        btn_query=findViewById(R.id.btn_query);
        btn_delete=findViewById(R.id.btn_delete);
        btn_insert=findViewById(R.id.btn_insert);
        btn_update=findViewById(R.id.btn_update);
        listView=findViewById(R.id.listView);

        adapter=new ArrayAdapter<>(this,android.R.layout.simple_list_item_1,items;
        listView.setAdapter(adapter);
        dbrw=new MyDBHelper(this).getWritableDatabase();
        btn_insert.setOnClickListener(view -> {
            if(ed_book.length()<1 || ed_price.length()<1){
                Toast.makeText(MainActivity.this,"欄位請勿留空",Toast.LENGTH_SHORT).show();
            }else{
                try{
                    dbrw.execSQL("INSERT INTO myTable(book,price) values(?,?)",new Object[]{ed_book.getText().toString(),ed_price.getText().toString()});
                    Toast.makeText(MainActivity.this,"成功書名"+ed_book.getText().toString()+"價格"+ed_price.getText().toString(),Toast.LENGTH_SHORT).show();
                    ed_book.setText("");
                    ed_price.setText("");
                }catch(Exception e){
                    e.printStackTrace();
                    Toast.makeText(MainActivity.this,"新增失敗"+e.getMessage(),Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: Lab13_withKotlin/app/src/main/java/com/stella/lab13_withkotlin/MainActivity.kt
package com.stella.lab13_withkotlin

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    lateinit var tv_clock: TextView
    lateinit var btn_start: Button
    private var flag = false
    private val receiver: BroadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            val b = intent.extras
            tv_clock!!.text =
                String.format("%02d:%02d:%02d", b!!.getInt("H"), b.getInt("M"), b.getInt("S"))
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tv_clock = findViewById(R.id.tv_clock)
        btn_start = findViewById(R.id.btn_start)
        registerReceiver(receiver, IntentFilter("MyMessage"))
        flag = MyService.flag
        if (flag) btn_start.setText("暫停") else btn_start.setText("開始")
        btn_start.setOnClickListener(View.OnClickListener {
            flag = !flag
            if (flag) {
                btn_start.setText("暫停")
                Toast.makeText(this@MainActivity, "計時開始", Toast.LENGTH_SHORT).show()
            } else {
                btn_start.setText("開始")
                Toast.makeText(this@MainActivity, "計時暫停", Toast.LENGTH_SHORT).show()
            }
            startService(Intent(this@MainActivity, MyService::class.java).putExtra("flag", flag))
        })
    }

    public override fun onDestroy() {
        super.onDestroy()
        unregisterReceiver(receiver)
    }
}